### PR TITLE
add codemod for icon swapping

### PR DESCRIPTION
**Pull request by mshwery**
This PR introduces a codemod for transition from `<Icon icon="foo" />` to tree-shakeable imports like `<FooIcon />`

It relies on jest for tests, so we may consider switching from ava to jest.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -58,7 +59,7 @@
     "fuzzaldrin-plus": "^0.6.0",
     "glamor": "^2.20.40",
     "lodash.debounce": "^4.0.8",
-    "lodash.mapvalues": "^4.6.0",
```
Comment (mshwery) [inline package.json:61]: unused

File: .prettierignore [hunk 2]
```diff
@@ -1 +1,2 @@
 package.json
+/codemods/
```
Comment (akleiner2) [inline .prettierignore:2]: Any reason we _don't_ want to have prettier run on the codemods dir?

File: package.json (hunk 1 above)
Comment (akleiner2) [reply package.json:61]: 👍

File: .prettierignore (hunk 2 above)
Comment (mshwery) [reply .prettierignore:2]: Yes – well mainly the test fixtures – I couldn't get the test runner to apply our prettier/babel config so the output matched properly. Any tips here?

**Pull request merged**